src/Event/Event.php

```diff
@@ -374,6 +376,49 @@ public function updateSubEventsStatus(array $statuses): void
         }
     }
 
+    public function updateSubEvents(SubEventUpdate ...$subEventUpdates): void
+    {
+        $timestamps = $this->calendar->getTimestamps();
+
+        if (empty($timestamps)) {
+            throw CalendarTypeNotSupported::forCalendarType($this->calendar->getType());
+        }
+
+        foreach ($subEventUpdates as $subEventUpdate) {
+            $index = $subEventUpdate->getSubEventIndex();
+
+            if (!isset($timestamps[$index])) {
+                // If the timestamp to update doesn't exist, it's most likely a concurrency issue.
+                continue;
+            }
+
+            $timestamp = $timestamps[$index];
+
+            $updatedTimestamp = new Timestamp(
+                $timestamp->getStartDate(),
+                $timestamp->getEndDate(),
+                $subEventUpdate->getStatus() ?: $timestamp->getStatus(),
+                $subEventUpdate->getBookingAvailability() ?: $timestamp->getBookingAvailability()
+            );
+
+            $timestamps[$index] = $updatedTimestamp;
+        }
+
+        $updatedCalendar = new Calendar(
+            $this->calendar->getType(),
+            $this->calendar->getStartDate(),
+            $this->calendar->getEndDate(),
```

Comment:
Nit: I think the startDate and endDate can be `null` if the `Calendar` has subEvents/timestamps, because the `Calendar` will then calculate them automatically anyway? Afaik `startDate` and `endDate` only need to be specified explicitly in the constructor for `periodic` calendars.

Reply:
Good point, makes it already easier for the update of start and end date also.

Reply:
This seems to have a side-effect. The accessor methods `getStartDate` and `getEndDate` do calculate it based on the time stamps. But internal code directly uses the `startDate` and `endDate` properties which are empty.

Reply:
Only an issue with the tests creating the `CalendarUpdated` event but not on the projectors itself. So fixes the failing tests and the rest is fine as it is now.